homestar-runtime/src/worker.rs

```diff
@@ -562,66 +562,50 @@ mod test {
-        let get_workflow_providers_event = rx.recv_async().await.unwrap();
-
-        // we should have received 2 receipts
-        let next_run_receipt = rx.recv_async().await.unwrap();
-        let next_next_run_receipt = rx.recv_async().await.unwrap();
-
-        match workflow_info_event {
-            Event::FindRecord(QueryRecord { cid, .. }) => assert_eq!(cid, worker_workflow_cid),
-            _ => panic!("Wrong event type"),
-        };
-
-        match get_workflow_providers_event {
-            Event::GetProviders(QueryRecord { cid, .. }) => assert_eq!(cid, worker_workflow_cid),
-            _ => panic!("Wrong event type"),
-        }
-
-        let (next_receipt, _wf_info) = match next_run_receipt {
-            Event::CapturedReceipt(Captured {
-                receipt: next_receipt,
-                ..
-            }) => {
-                let stored = workflow::Stored::default(Pointer::new(workflow_cid), 2);
-                let mut info = workflow::Info::default(stored);
-                info.increment_progress(next_receipt);
-
-                (next_receipt, info)
-            }
-            _ => panic!("Wrong event type"),
-        };
-
-        let (_next_next_receipt, wf_info) = match next_next_run_receipt {
-            Event::CapturedReceipt(Captured {
-                receipt: next_next_receipt,
-                ..
-            }) => {
-                let stored = workflow::Stored::default(Pointer::new(workflow_cid), 2);
-                let mut info = workflow::Info::default(stored);
-                info.increment_progress(next_next_receipt);
-
-                assert_ne!(next_next_receipt, next_receipt);
+        let mut find_record = false;
+        let mut get_providers = false;
+        let mut captured_receipt = false;
+        let mut receipts_cnt = 0;
 
-                (next_next_receipt, info)
+        while let Ok(event) = rx.recv_async().await {
+            match event {
+                Event::FindRecord(QueryRecord { cid, .. }) => {
+                    find_record = true;
+                    assert_eq!(cid, worker_workflow_cid)
+                }
+                Event::GetProviders(QueryRecord { cid, .. }) => {
+                    get_providers = true;
+                    assert_eq!(cid, worker_workflow_cid)
+                }
+                Event::CapturedReceipt(Captured { receipt, .. }) => {
+                    let stored = workflow::Stored::default(Pointer::new(workflow_cid), 2);
+                    let mut info = workflow::Info::default(stored);
+                    info.increment_progress(receipt);
+                    let (_, workflow_info) =
+                        MemoryDb::get_workflow_info(workflow_cid, &mut conn).unwrap();
+                    assert_eq!(info.progress_count, workflow_info.progress_count);
+                    if receipts_cnt == 1 {
+                        captured_receipt = true;
+                    } else {
+                        receipts_cnt += 1;
```

Comment:
Maybe don't need the conditional here? The receipts count can be incremented on each event and no harm is setting `captured_receipt` to true multiple times.